tests.py

```diff
@@ -46,6 +48,15 @@ def __init__(self):
     def update(self, responses):
         self.responses.update(responses)
 
+    def set_user(self, user):
+        if user is None:
+            if self.user_endpoint in self.responses:
+                del self.responses[self.user_endpoint]
+        else:
+            self.responses[self.user_endpoint] = MockResponse(
+                json={'login': user},
+            )
```

Comment:
I'd like to make the mock a bit smarter and return a response only when the request contains an Authorization header, failing with a 401 otherwise.

I think that would have caught the use case where you're now calling _verify_user_token() even if the user hasn't provided a token.

Reply:
So if the user is None, the response should be 401 instead of failing with an error because the mock wasn't completely configured?

Reply:
I've just checked: `get_json_and_links()` converts all 4xx errors to an exception, so I don't think the distinction between 401 and 404 really matters for these tests.